.github/workflows/set-date.yaml

```diff
@@ -0,0 +1,27 @@
+name: Set Date
+
+on:
+  schedule:
+    - cron: '* 0 * * *'
+
+jobs:
+  update-date:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: Write Date File
+      run: |
+        echo -e -n "#ifndef VCS_DATE\n#define VCS_DATE $(date "+%Y%m%d")\n#endif" > ./firmware/controllers/date_stamp.h
```

Comment:
minor simplification - can we use "#pragma once" here? ifndef VCS_DATE hints at a possibility to externally define it which we do not need. "#pragma once" would close that door thus making the thing overall easier?